### PR TITLE
modify _playWAVFile to also support 16bit WAV files

### PR DESCRIPTION
**Pull request by simonedegiacomi**
This PR allows `_playWAVFile` to play 16bit WAV files.

I recognize that this feature may not be commonly needed, however I use it to play synthesized phrases generate on the robot using espeak, and I thought that maybe it could be useful also to other users.

### CODE REVIEW

File: API/ev3_sound.c [hunk 1]
```diff
@@ -322,7 +322,28 @@ void _playRMDFile(char* pFileName, uint8_t volume, bool loop)
...
+#define RIFF_FMT_8BITS  0x0800
+#define RIFF_FMT_16BITS 0x1000
+#define RIFF_DATA_SIG   0x64617461
+
+int changeBitDepthFrom16To8 (uint8_t buffer[], int length) {
+	int i, j = 0;
+	for (i = 0; i < length; i += 2) {
+		uint8_t b1 = buffer[i];
+		uint8_t b2 = buffer[i + 1];
+		short tmp = (b1 & 0xff) | ((b2 & 0xff) << 8); // from 2 bytes to short
```
Comment (a3f) [inline API/ev3_sound.c:340]: No need to mask, `b1` and `b2` are already `<= 0xff`

File: API/ev3_sound.c [hunk 2]
```diff
@@ -322,7 +322,28 @@ void _playRMDFile(char* pFileName, uint8_t volume, bool loop)
...
+		buffer[j++] = ((tmp + 32767) >> 8) & 0xff;
```
Comment (a3f) [inline API/ev3_sound.c:341]: Signed integer overflow in `tmp` is undefined behavior.

**Review by a3f — CHANGES_REQUESTED** [review]

**Comment by a3f** [comment]
> I recognize that this feature may not be commonly needed, however I use it to play synthesized phrases generate on the robot using espeak, and I thought that maybe it could be useful also to other users.

As long it doesn't pull in extra dependencies, I don't mind pulling it in.

File: API/ev3_sound.c (hunk 2 above)
Comment (simonedegiacomi) [reply API/ev3_sound.c:341]: If I understood it correctly, the problem appears when if `b1+b2+32767` is greater than `SHORT_MAX`, correct?

If that's correct, would casting `tmp` to an int before doing the sum solve the problem?

```c
short tmp = b1 | (b2 << 8);
buffer[j++] = ((((int)tmp) + 32767) >> 8) & 0xff;
```

Comment (a3f) [reply API/ev3_sound.c:341]: Never mind, C implicitly converts the `tmp` to an `int` anyway. No overflow here as long as `2*sizeof(short) == sizeof(int)`

**Review by a3f — APPROVED** [review]

**Comment by a3f** [comment]
Applied, thanks!

**Pull request merged**